### DISCUSSION TITLE
getStaticProps / getInitialProps output & SEO

### DISCUSSION THREAD

**Discussion by pateljoel** [vercel/next.js#12647] [Help]
Thanks for building Next.js, currently using it in a project which has tons of data and saw the SEO benefits of it through the official tutorial.  

I used the new `getStaticProps()` (previously `getInitialProps()`) function, to pull JSON data at build time, statically generating hundreds of pages. 

I had a look a the HTML that had been generated and saw a variable called `__NEXT_DATA__`, which the props is huge, around 60% of the size of my HTML page.

Having looked at https://github.com/zeit/next.js/issues/10332 https://github.com/zeit/next.js/issues/2795 and https://github.com/zeit/next.js/issues/1637 I understand that there is no plans to remove or change `__NEXT_DATA__` because it's necessary, if possible would like to see if there are ways around it.

If not, how does the `__NEXT_DATA__` impact SEO? Does Google crawl this data and take it into account? 

Thanks

**Comment by revskill10** [comment]
You can remove those script in `_document.js` ? I mean in case we only need static html without hydration for interaction.
It would be helpful if Next.js could allow configure things like `isStatic: true/false` in `getStaticProps`, so if `isStatic  is true`, the output will trip out all `<script>`, there's only html left. My 2c.

**Comment by DennieMello** [comment]
Also wondering, can someone have an implementation of loading __NEXT_DATA_ via json file?

**Reply by leerob** [reply]
Out of curiosity, why would you want this?

**Comment by leerob** [comment]
> Having looked at #10332 #2795 and #1637 I understand that there is no plans to remove or change __NEXT_DATA__ because it's necessary, if possible would like to see if there are ways around it.

Correct, [this comment](https://github.com/vercel/next.js/issues/2795#issuecomment-471213936) explains why. It sounds like there's a potential workaround with Redux, if you're using that. There's also an experimental flag to remove all client-side JS, but use this at your own risk.

```js
export const config = {
  unstable_runtimeJS: false
};
```

> If not, how does the __NEXT_DATA__ impact SEO?

It doesn't impact SEO negatively 👍  Side note: using pre-rendering with Next.js _improves_ SEO because search engines can crawl the HTML content immediately.

**Reply by pateljoel** [reply]
This sounds like a good workaround, but this a page wide thing? I think there should be some docs on this.

**Reply by leerob** [reply]
In general, we don't add official docs for workarounds. Docs would be added once official support lands, if necessary.

**Comment by timhysniu** [comment]
Next is amazing about many things, but some of us have chosen to use Next primarily for SEO benefits. Is there no setting that can force rendering of HTML so that we're not looking at `__NEXT_DATA__` in a page. I don't think this will be very useful to Google Bot.

Interestingly, I do see that some pages do render content without `__NEXT_DATA__`. Does this depend on size of content perhaps?

**Comment by ghost** [comment]
I think the __NEXT_DATA__   is not a good idea  some times, because  __NEXT_DATA__  will output all  fields of data structer, even some private  fields just like token/id number/order number and so on...

**Reply by GabenGar** [reply]
Well it is a file sent to client, how exactly do you expect privacy there?
